src/classes/SObjectRepository.cls

```diff
@@ -2,40 +2,102 @@
+    }
+
+    protected IQueryBuilder queryBuilder {
+        get {
+            return new QueryBuilder(sObjectType, fieldSet, sObjectFieldList);
+        }
+    }
+
+    public IQueryBuilder getQueryBuilder() {
+        return this.queryBuilder;
+    }
+    //SOQL
+    public virtual SObject getById(Id recordId) {
+        return this.queryBuilder
+            .whereField(new List<QueryPredicate>{QueryPredicate.EQUALS(this.idField, recordId)})
+            .getFirstOrDefault();
+    }
+
+    public virtual List<SObject> getById(List<Id> recordIds) {
+        return this.queryBuilder
+            .whereField(new List<QueryPredicate>{QueryPredicate.EQUALS(this.idField, recordIds)})
+            .getQueryResults();
+    }
+
+    public virtual List<SObject> get(List<QueryPredicate> queryPredicates) {
+        return this.queryBuilder
+            .whereField(queryPredicates)
+            .getQueryResults();
+    }
+
+    public List<SObject> get(QueryPredicate queryPredicate) {
+        return this.get(new List<QueryPredicate> { queryPredicate });
     }
 
-    public virtual void doInsert(SObject record) {this.doInsert(new List<SObject>{record});}
-    public virtual void doInsert(List<SObject> recordList) {Database.insert(recordList);}
-    public virtual void doUpdate(SObject record) {this.doUpdate(new List<SObject>{record});}
-    public virtual void doUpdate(List<SObject> recordList) {Database.update(recordList);}
-    // Apex doesn't allow upserting on generic SObject, so we'll delegate the implementation to each concrete class
-    public abstract void doUpsert(SObject record);
-    public abstract void doUpsert(List<SObject> recordList);
 
-    public virtual void doDelete(SObject record) {this.doDelete(new List<SObject>{record});}
-    public virtual void doDelete(List<SObject> recordList) {Database.delete(recordList);}
-    public virtual void doHardDelete(SObject record) {this.doHardDelete(new List<SObject>{record});}
-    public virtual void doHardDelete(List<SObject> recordList) {
-        Database.delete(recordList);
-        if(!recordList.isEmpty()) Database.emptyRecycleBin(recordList);
+    public virtual List<SObject> getByFieldForIds(Schema.SObjectField field, Object value, Set<Id> idSet) {
+        return this.getByFieldForIds(field, value, new List<Id>(idSet));
     }
-    public virtual void doUndelete(SObject record) {this.doUndelete(new List<SObject>{record});}
-    public virtual void doUndelete(List<SObject> recordList) {Database.undelete(recordList);}
 
+    public virtual List<SObject> getByFieldForIds(Schema.SObjectField field, Object value, List<Id> recordIds) {
+        return this.queryBuilder
+            .whereField(this.idField, QueryOperator.IS_IN, recordIds)
+            .whereField(field, QueryOperator.EQUALS, value)
+            .getQueryResults();
+    }
+
+    public List<SObject> getCreatedSinceTimeValue(Object timeValue) {
+        Schema.SObjectField createdDate = this.getCreatedDate();
```

Comment:
I'd scrap this line (and the variable) and just call the method directly in the whereField call
`.whereField(this.getCreatedDate(), QueryOperator.GREATER_THAN_OR_EQUAL_TO, timeValue)`